pkg/storage/storage_fs.go

```diff
@@ -159,8 +159,23 @@ func (is *ImageStoreFS) Unlock() {
 
 func (is *ImageStoreFS) initRepo(name string) error {
 	repoDir := path.Join(is.rootDir, name)
+
+	absRepoDir, err := filepath.Abs(repoDir)
+
+	if err != nil {
+		is.log.Error().Err(err).Msg("error trying to get absolute path of repo")
+
+		return err
+	}
+
+	if (is.rootDir + "/" + name) != absRepoDir {
```

Comment:
use path.Join(is.rootDir, name)

Reply:
I could have used this in the first place and no changes would have been needed.

The problem is what if name = "../../abc" the join would be different from what we expect.